src/reducers/versions.spec.tsx

```diff
@@ -2470,4 +2524,206 @@ describe(__filename, () => {
+          // to set up a location in the router in the test.
+          expect.objectContaining({
+            hash: `#${anchor}`,
+          }),
+        ),
+      );
+    });
+
+    it.each([
+      ['next', RelativePathPosition.next, ScrollTarget.firstDiff],
+      ['previous', RelativePathPosition.previous, ScrollTarget.lastDiff],
+    ])(
+      'dispatches viewVersionFile for a new file with "%s"',
+      async (direction, position, diffPosition) => {
+        const path = 'newFile.js';
+        const _getRelativeDiff = jest.fn().mockReturnValue({ path });
+        const currentAnchor = '';
+        const diff = null;
+        const pathList = ['file1.js'];
+        const versionId = 123;
+
+        const fakeThunk = createFakeThunk();
+        const _viewVersionFile = fakeThunk.createThunk;
+
+        const store = configureStore();
+        store.dispatch(
+          actions.loadVersionInfo({
+            version: { ...fakeVersion, id: versionId },
+          }),
+        );
+
+        const typedPosition = position as RelativePathPosition;
+
+        const { dispatch, thunk } = thunkTester({
+          createThunk: () =>
+            _goToRelativeDiff({
+              _getRelativeDiff,
+              _viewVersionFile,
+              currentAnchor,
+              diff,
+              pathList,
+              position: typedPosition,
+              versionId,
+            }),
+          store,
+        });
+
+        await thunk();
+
+        expect(dispatch).toHaveBeenCalledWith(fakeThunk.thunk);
+        expect(_viewVersionFile).toHaveBeenCalledWith({
+          preserveHash: false,
+          selectedPath: path,
+          showDiff: diffPosition,
+          versionId,
+        });
+      },
+    );
+
+    it('dispatches nothing if no anchor or path is returned', async () => {
+      const _getRelativeDiff = jest
+        .fn()
+        .mockReturnValue({ anchor: null, path: null });
+      const versionId = 123;
+
+      const store = configureStore();
+      store.dispatch(
+        actions.loadVersionInfo({ version: { ...fakeVersion, id: versionId } }),
+      );
+
+      const { dispatch, thunk } = thunkTester({
+        createThunk: () =>
+          _goToRelativeDiff({
+            _getRelativeDiff,
+            versionId,
+          }),
+        store,
+      });
+
+      await thunk();
+
+      expect(dispatch).not.toHaveBeenCalled();
+    });
+
+    it('throws an exception if the version is not loaded', async () => {
+      const versionId = 123;
+      const store = configureStore();
+      store.dispatch(
+        actions.loadVersionInfo({ version: { ...fakeVersion, id: versionId } }),
+      );
+
+      const { thunk } = thunkTester({
+        createThunk: () =>
+          _goToRelativeDiff({
+            versionId: versionId + 1,
+          }),
+        store,
+      });
+
+      // TODO: This doesn't work. Maybe we cannot test for an exception in a thunk?
+      expect(async () => {
+        await thunk();
```

Comment:
`await` is just syntactic sugar to work with promises. In this case you could either write your own `try/catch` block around `await thunk()` or deal with the promise directly, like this, which is a bit easier:

```diff
diff --git a/src/reducers/versions.spec.tsx b/src/reducers/versions.spec.tsx
index 76e5fef..0e9e923 100644
--- a/src/reducers/versions.spec.tsx
+++ b/src/reducers/versions.spec.tsx
@@ -2692,9 +2692,9 @@ describe(__filename, () => {
       });
 
       // TODO: This doesn't work. Maybe we cannot test for an exception in a thunk?
-      expect(async () => {
-        await thunk();
-      }).toThrow('Cannot go to relative diff without a version loaded');
+      await expect(thunk()).rejects.toThrow(
+        'Cannot go to relative diff without a version loaded',
+      );
     });
 
     it('does not call _getRelativeDiff if the version is not loaded', async () => {

```

Reply:
Ah, I was trying to find a `toReject` function similar to `toThrow`, but I didn't think to look for `rejects`.